airbyte-webapp/src/components/ui/NextTable/NextTable.tsx

```diff
@@ -1,43 +1,72 @@
 import classNames from "classnames";
 import { PropsWithChildren } from "react";
 
 import styles from "./NextTable.module.scss";
 
-export interface TableProps<TData> {
-  data: TData[];
-  columns: Array<ColumnDef<TData>>;
-  onClickRow?: (data: unknown) => void;
+export interface TableProps<T> {
   className?: string;
+  // We can leave type any here since useReactTable options.columns itself is waiting for Array<ColumnDef<T, any>>
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  columns: Array<ColumnDef<T, any>>;
+  data: T[];
+  erroredRows?: boolean;
+  light?: boolean;
+  onClickRow?: (data: T) => void;
+  columnSort?: ColumnSort[];
+  testId?: string;
 }
 
-export const NextTable = <TData,>({ columns, data, onClickRow, className }: PropsWithChildren<TableProps<TData>>) => {
+export const NextTable = <T,>({
+  testId,
+  className,
+  columns,
+  data,
+  erroredRows,
+  light,
+  onClickRow,
+}: PropsWithChildren<TableProps<T>>) => {
   const table = useReactTable({
     columns,
     data,
     getCoreRowModel: getCoreRowModel(),
   });
 
   return (
-    <table className={classNames(styles.table, className)}>
+    <table className={classNames(styles.table, className, { [styles.light]: light })} data-testid={testId}>
       <thead className={styles.thead}>
         {table.getHeaderGroups().map((headerGroup) => (
           <tr key={`table-header-${headerGroup.id}}`}>
-            {headerGroup.headers.map((header) => (
-              <th
-                colSpan={header.colSpan}
-                className={classNames(styles.th, header.column.columnDef.meta?.thClassName)}
-                key={`table-column-${headerGroup.id}-${header.id}`}
-              >
-                {flexRender(header.column.columnDef.header, header.getContext())}
-              </th>
-            ))}
+            {headerGroup.headers.map((header) => {
+              const { meta } = header.column.columnDef;
+              return (
+                <th
+                  colSpan={header.colSpan}
+                  className={classNames(styles.th, meta?.thClassName, {
+                    [styles.highlighted]: meta?.headerHighlighted,
+                    [styles.light]: light,
+                  })}
+                  key={`table-column-${headerGroup.id}-${header.id}`}
+                >
+                  {flexRender(header.column.columnDef.header, header.getContext())}
+                </th>
+              );
+            })}
           </tr>
         ))}
       </thead>
       <tbody>
         {table.getRowModel().rows.map((row) => (
-          <tr className={styles.tr} key={`table-row-${row.id}`} onClick={() => onClickRow?.(row.original)}>
+          <tr
+            className={classNames(styles.tr, {
+              // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+              // @ts-ignore
+              [styles.withError]: erroredRows && !!row.original.error,
```

Comment:
I can't find a code path where we would need/want this.  I may be wrong here... but I _think_ this may be removable.

Reply:
+1 agree